mtp_api/apps/transaction/tests/test_views.py

```diff
@@ -7,105 +7,744 @@
+
+        # check 5 taken transactions in db
+        self.assertEqual(
+            self._get_pending_transactions_qs(prison, owner).count(),
+            count
+        )
+
+    def test_nobody_else_can_take_transactions_for_others(self):
+        """
+        Anybody managing a prison should be able to take transactions
+        on behalf of other users within the same prison.
+        """
+        prison = self.prisons[0]
+        users = get_users_for_prison(prison)
+        self.assertTrue(len(users) >= 2)
+
+        # We need 2 users as we want to test that logged_in_user
+        # can take transactions on behalf of transactions_owner.
+        # As result, the transactions.owner should be transactions_owner
+        # and not logged_in_user
+        logged_in_user = users[0]
+        transactions_owner = users[1]
+
+        # delete pending transactions just to clean things up
+        self._get_pending_transactions_qs(prison, transactions_owner).delete()
+
+        # check no taken transactions in db
+        self.assertEqual(
+            self._get_pending_transactions_qs(prison, transactions_owner).count(),
+            0
+        )
+
+        # request
+        self.client.force_authenticate(user=logged_in_user)
+
+        url = self._get_url(transactions_owner, prison, 1)
+        response = self.client.post(url, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        # check 5 taken transactions in db
+        self.assertEqual(
+            self._get_pending_transactions_qs(prison, transactions_owner).count(),
+            0
+        )
+
+    def test_fails_when_taking_too_many(self):
+        """
+        Tests that fails when trying to take more than TAKE_LIMIT.
+        """
+        user = self.owners[0]
+        prison = get_prisons_for_user(user)[0]
+
+        # clean things up
+        self._get_pending_transactions_qs(prison).update(owner=None)
+
+        # make sure we have enough available transactions
+        self.assertTrue(
+            self._get_available_transactions_qs(prison).count() > TAKE_LIMIT
+        )
+
+        # check no taken transactions in db
+        self.assertEqual(
+            self._get_pending_transactions_qs(prison, user).count(), 0
+        )
+
+        # request TAKE_LIMIT-1
+        count = TAKE_LIMIT-1
+        self.client.force_authenticate(user=user)
+
+        url = self._get_url(user, prison, count)
+        response = self.client.post(url, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_303_SEE_OTHER)
+
+        # check TAKE_LIMIT-1 taken transactions in db
+        self.assertEqual(
+            self._get_pending_transactions_qs(prison, user).count(), count
+        )
+
+        # request 2 more => should fail
+        url = self._get_url(user, prison, 2)
+        response = self.client.post(url, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
I'm happy enough with it being as it is, but I think your suggestion of 409 and a suggestion to release some would be nice?